File: ui/windows/ManifestWriter.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.IO;
using System.Text;

namespace AgentVideoParse
{
    internal static class ManifestWriter
    {
        public static string FrameFilename(int index)
        {
            return string.Format(CultureInfo.InvariantCulture,
                "frame-{0:D4}.{1}", index, AvpConstants.FrameExtension);
        }

        public static string WriteManifest(
            string outputDirectory,
            string sourcePath,
            double durationSeconds,
            IList<Tuple<int, double, string>> entries,
            double sampleFps = AvpConstants.DefaultSampleFps)
        {
            string path = Path.Combine(outputDirectory, "MANIFEST.txt");
            string generated = DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture);
            var sb = new StringBuilder();
            sb.AppendLine("# AgentVideoParse manifest");
            sb.AppendLine("# purpose: debugging / agent UI review only");
            sb.AppendLine("# source: " + sourcePath);
            sb.AppendLine(string.Format(CultureInfo.InvariantCulture, "# duration_seconds: {0:F6}", durationSeconds));
            sb.AppendLine(string.Format(CultureInfo.InvariantCulture, "# max_allowed_seconds: {0:g}", AvpConstants.DurationLimitSeconds));
            sb.AppendLine(string.Format(CultureInfo.InvariantCulture, "# sample_fps: {0:g}", sampleFps));
            sb.AppendLine("# frame_count: " + entries.Count);
            sb.AppendLine("# max_frames: " + AvpConstants.MaxFrames);
            sb.AppendLine("# image_format: " + AvpConstants.FrameExtension);
            sb.AppendLine("# max_long_edge: " + AvpConstants.MaxLongEdge);
            sb.AppendLine("# jpeg_quality: " + AvpConstants.JpegQuality);
            sb.AppendLine("# platform: windows-app");
            sb.AppendLine("# generated_at: " + generated);
            sb.AppendLine();
            sb.AppendLine("index\ttimestamp_seconds\tfilename");
            foreach (var e in entries)
            {
                sb.AppendLine(string.Format(CultureInfo.InvariantCulture,
                    "{0}\t{1:F3}\t{2}", e.Item1, e.Item2, e.Item3));
            }
            File.WriteAllText(path, sb.ToString(), new UTF8Encoding(false));
            return path;
        }

        public static string WriteAgentReadme(string outputDirectory)
        {
            string path = Path.Combine(outputDirectory, "README-FOR-AGENT.txt");
            string text =
                "AgentVideoParse output\n" +
                "======================\n\n" +
                "These ordered screenshots were extracted from a short debug video " +
                "(maximum 60 seconds). This folder is for AI/agent UI debugging only.\n\n" +
                "Read MANIFEST.txt for index → timestamp → filename mapping.\n" +
                "Frames are named frame-0001." + AvpConstants.FrameExtension +
                ", … in time order (agent-friendly JPEG, long edge ≤ " +
                AvpConstants.MaxLongEdge + "px).\n";
            File.WriteAllText(path, text, new UTF8Encoding(false));
            return path;
        }
    }
}

File: macos/AgentVideoParse/Sources/ManifestWriter.swift
import Foundation

enum ManifestWriter {
    static func frameFilename(index: Int) -> String {
        String(format: "frame-%04d.%@", index, AVPConstants.frameExtension)
    }

    static func writeManifest(
        outputDirectory: URL,
        sourcePath: String,
        durationSeconds: Double,
        entries: [(index: Int, timestamp: Double, filename: String)],
        sampleFPS: Double = AVPConstants.defaultSampleFPS
    ) throws -> URL {
        let path = outputDirectory.appendingPathComponent("MANIFEST.txt")
        let generated = ISO8601DateFormatter().string(from: Date())
        var lines: [String] = [
            "# AgentVideoParse manifest",
            "# purpose: debugging / agent UI review only",
            "# source: \(sourcePath)",
            String(format: "# duration_seconds: %.6f", durationSeconds),
            "# max_allowed_seconds: \(AVPConstants.durationLimitSeconds)",
            "# sample_fps: \(sampleFPS)",
            "# frame_count: \(entries.count)",
            "# max_frames: \(AVPConstants.maxFrames)",
            "# image_format: \(AVPConstants.frameExtension)",
            "# max_long_edge: \(AVPConstants.maxLongEdge)",
            "# jpeg_quality: \(AVPConstants.jpegQuality)",
            "# platform: macos-app",
            "# generated_at: \(generated)",
            "",
            "index\ttimestamp_seconds\tfilename",
        ]
        for e in entries {
            lines.append(String(format: "%d\t%.3f\t%@", e.index, e.timestamp, e.filename))
        }
        try lines.joined(separator: "\n").appending("\n").write(to: path, atomically: true, encoding: .utf8)
        return path
    }

    static func writeAgentReadme(outputDirectory: URL) throws -> URL {
        let path = outputDirectory.appendingPathComponent("README-FOR-AGENT.txt")
        let text = """
        AgentVideoParse output
        ======================

        These ordered screenshots were extracted from a short debug video \
        (maximum 60 seconds). This folder is for AI/agent UI debugging only.

        Read MANIFEST.txt for index → timestamp → filename mapping.
        Frames are named frame-0001.jpg, frame-0002.jpg, … in time order
        (agent-friendly JPEG, long edge ≤ \(AVPConstants.maxLongEdge)px).
        """
        try text.write(to: path, atomically: true, encoding: .utf8)
        return path
    }
}
